index.js

```diff
@@ -30,56 +48,77 @@ module.exports = options => {
 			return;
 		}
 
-		const finish = (error, size) => {
+		const finish = (error, sizes) => {
 			if (error) {
 				callback(new PluginError('gulp-size', error));
 				return;
 			}
 
-			totalSize += size;
+			sizes.forEach((size, key) => totalSize.set(key, size + (totalSize.get(key) || 0)));
 
-			if (options.showFiles === true && size > 0) {
-				log(chalk.blue(file.relative), size);
+			if (options.showFiles === true && hasSize(sizes)) {
+				log(chalk.blue(file.relative), sizes);
 			}
 
 			fileCount++;
 			callback(null, file);
 		};
 
+		const calc = [];
+		const names = [];
```

Comment:
Unless I'm missing something, it seems better to use a `Map` here than two arrays.

Reply:
I don't think you're missing something.